Plugin/WebapiRest/Route/Route/Helper/DataPlugin.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2022 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Plugin\WebapiRest\Route\Route\Helper;
+
+/**
+ * Plugin for {@see \Route\Route\Helper\Data}
+ */
+class DataPlugin
+{
+    public const CART_ID_PARAM = 'cartId';
+    public const IS_INSURED_PARAM = 'route_is_insured';
+    /**
+     * @var \Magento\Framework\Webapi\Rest\Request
+     */
+    private $request;
+
+    /**
+     * @var \Magento\Quote\Api\CartRepositoryInterface
+     */
+    private $quoteRepository;
+
+    /**
+     * @var \Bolt\Boltpay\Model\ThirdPartyModuleFactory
+     */
+    private $boltRouteQuoteClient;
+
+    public function __construct(
+        \Magento\Framework\Webapi\Rest\Request $request,
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        \Bolt\Boltpay\Model\ThirdPartyModuleFactory $boltRouteQuoteClient
+    ){
+        $this->request = $request;
+        $this->quoteRepository = $quoteRepository;
+        $this->boltRouteQuoteClient = $boltRouteQuoteClient;
+    }
+
+    public function afterIsInsured(\Route\Route\Helper\Data $subject, $result): bool
+    {
+        $requestParams = $this->request->getParams();
```

Comment:
Finding parameter in request is a little bit counterintuitive.
What do you think about having a set method in this class, like SetIsInsured, and call it directly from API class?
This is also a way to show dependency between API class and this plugin